network/endpoint_windows.go

```diff
@@ -14,16 +14,58 @@ import (
 	"github.com/Microsoft/hcsshim"
 )
 
+// ConstructEpName constructs endpoint name from netNsPath.
+func ConstructEpName(containerID string, netNsPath string, ifName string) string {
+	epName := ""
+	if netNsPath != "" {
+		splits := strings.Split(netNsPath, ":")
+		if len(splits) == 2 {
+			epName = splits[1]
+		} else {
+			epName = containerID
+		}
+		if len(epName) > 8 {
+			epName = epName[:8] + "-" + ifName
+			log.Printf("[net] constructed epName: %v", epName)
+		}
+	}
+	return epName
+}
+
 // newEndpointImpl creates a new endpoint in the network.
 func (nw *network) newEndpointImpl(epInfo *EndpointInfo) (*endpoint, error) {
+	// Ignore consecutive ADD calls for the same container.
+	if nw.Endpoints[epInfo.Id] != nil {
```

Comment:
Are we using epName or epinfo.Id as a key in Endpoints map? In line 54, we are using epName as key for same map? are both same?

Reply:
epinfo.Id is generated from containerID of ADD call. If the ADD call is for infrastructure container, then epName == epinfo.id. Basic, when I use epName as a key for nw.Endpoints, I am handling the ADD call for workload container. I want to return the endpoint for infrastructure container, since these two containers belong to the same pod. Reference: https://kubernetes.io/docs/concepts/cluster-administration/networking/#kubernetes-model

Reply:
Thank you got it.  Checking GetHNSEndpointByName not nil solves object already exists issue. Can you please add some details in function description?